data-prepper-core/src/main/java/org/opensearch/dataprepper/parser/model/DataPrepperConfiguration.java

```diff
@@ -85,9 +86,11 @@ public DataPrepperConfiguration(
             final Duration sinkShutdownTimeout,
             @JsonProperty("circuit_breakers") final CircuitBreakerConfig circuitBreakerConfig,
             @JsonProperty("source_coordination") final SourceCoordinationConfig sourceCoordinationConfig,
+            @JsonProperty("kafka_cluster_config") final KafkaClusterConfig kafkaClusterConfig,
```

Comment:
This is a perfect place to use Data Prepper extensions.

See my comment on your issue: https://github.com/opensearch-project/data-prepper/issues/3233#issuecomment-1699806916